src/main/java/com/mraof/minestuck/entity/AnimatedCreatureEntity.java

```diff
@@ -0,0 +1,163 @@
+package com.mraof.minestuck.entity;
+
+import net.minecraft.entity.CreatureEntity;
+import net.minecraft.entity.EntityType;
+import net.minecraft.entity.LivingEntity;
+import net.minecraft.entity.ai.goal.MeleeAttackGoal;
+import net.minecraft.network.datasync.DataParameter;
+import net.minecraft.network.datasync.DataSerializers;
+import net.minecraft.network.datasync.EntityDataManager;
+import net.minecraft.util.Hand;
+import net.minecraft.world.World;
+import software.bernie.geckolib3.core.IAnimatable;
+import software.bernie.geckolib3.core.controller.AnimationController;
+import software.bernie.geckolib3.core.manager.AnimationFactory;
+
+public abstract class AnimatedCreatureEntity extends CreatureEntity implements IAnimatable {
+    private final AnimationFactory factory = new AnimationFactory(this);
+    protected static final DataParameter<Integer> CURRENT_ACTION = EntityDataManager.defineId(AnimatedCreatureEntity.class, DataSerializers.INT);
+
+    private int heavyAttackTicks = 0;
+    private int recoveryTicks = 0;
+    private int attackDelay;
+    private int attackRecovery;
+    private int actionResetTicks;
+
+    protected AnimatedCreatureEntity(EntityType<? extends CreatureEntity> type, World world) {
+        super(type, world);
+    }
+
+    @Override
+    protected void registerGoals() {
+        super.registerGoals();
+        goalSelector.addGoal(3, new DelayedAttackGoal(this, 1F, false));
+    }
```

Comment:
Unsure if it would be a goal, but add a TODO or implement head tracking so that entities look more natural and more frequently face the entity of their interest

Reply:
To reword and make it clear, I mean that code would be added so that the head bone(s) of entities can move with some independence to the rest of the animation. That way its not always facing the exact same way its body faces